backend/src/ee/services/secret-snapshot/secret-snapshot-service.ts

```diff
@@ -81,8 +84,9 @@ export const secretSnapshotServiceFactory = ({
     const folder = await folderDAL.findBySecretPath(projectId, environment, path);
     if (!folder) throw new BadRequestError({ message: "Folder not found" });
 
+    const project = await projectDAL.findById(projectId);
     const count = await snapshotDAL.countOfSnapshotsByFolderId(folder.id);
-    return count;
+    return Math.min(count, project.pitVersionLimit);
```

Comment:
We shouldn'tdo this and show the commits happening. Because this will make user think commits are not getting saved. 

We should make it clearr in ui of project settings that it gets deleted overnight

Reply:
addressed ^